### DISCUSSION TITLE
how to make a modal flow like twitter's tweet with dynamic route?

### DISCUSSION THREAD

**Discussion by iam-yan** [vercel/next.js#12904] [Help]
Hi, I'd like to create a modal flow like twitter's tweet:
1. Click tweet button in any post page, e.g. https://twitter.com/pitdesi/status/1260967133657174016
2. A new entry is pushed into history, but there is no refresh. Only the asPath is changed to https://twitter.com/compose/tweet   and a tweet modal is invoked.

What is the best way to achieve this?
I could only think of using 'catch all' route...

**Comment by jamesmosier** [comment]
[This discussion](https://github.com/zeit/next.js/discussions/12878) might be of some help, although I'm not certain it has been fully resolved.

Additionally here is [an example](https://github.com/zeit/next.js/tree/canary/examples/with-route-as-modal) or something similar.

**Reply by iam-yan** [reply]
@jamesmosier 
Hey thanks! I have checked the examples you gave yet I think they are different cases.

The problem in the twitter flow is that, the modal's path is in a higher level. Thus at some point, I'd have to do something like `<Link href="/post/[id]" as="/tweet">`, which would bring an "Incompatible href and as values" error.

**Comment by sambhav-gore** [comment]
I also want something similar.
In an Earlier project where we Did SSR on our own, this is how we achieved it (and I am going to try if it would work with NextJS in a few days). If you would like you can try this as well :

client:
1. When the modal is launched - use `window.history.pushState` to update the address.

ssr:

2. When someone comes directly from that route, Render the same thing that the default index route does (I haven't figured how this would be done but that are several options) 
3. IN the render tree look for specific path and Launch the modal based off that.

**Reply by carlbrn** [reply]
1. It breaks next-router's history.

**Comment by reloadlife** [comment]
have you found a way of doing this yet?

**Reply by carlbrn** [reply]
In case you still need it. It's possible. Are you using the app or the page router?

**Reply by reloadlife** [reply]
already did it using intercepting routes. thank you
